File: games/calculator.py
def add(a,b):
    return a+b

def subract(a,b):
    return a-b

def multiply(a,b):
    return a*b

def divide(a,b):
    return a/b

def exponentialize(base,exponent):
    if exponent == 0:
        return 1
    elif exponent < 0:
        return 1/exponentialize(base,abs(exponent))
    elif exponent > 0:
        result = 1
        for _ in range(exponent):
            result *= base
        return result
    
def root(num,index=2):
    if index > 0:
        return pow(num,1/index)
    else:
        return 'Invalid operation. Enter a positive index.'
    

File: games/calculator.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main()
{
    char op;
    double num1;
    double num2;
    std::cout << "Enter the operation (+, -, *, /, or ^): ";
    std::cin >> op;
    std::cout << "Enter the first number: ";
    std::cin >> num1;
    std::cout << "Enter the second number: ";
    std::cin >> num2;
    char eq = '=';
    switch (op)
    {
    case '+':
        std::cout << num1 << " " << op << " " << num2 << " " << eq << " " << (num1 + num2);
        break;
    case '-':
        std::cout << num1 << " " << op << " " << num2 << " " << eq << " " << (num1 - num2);
        break;
    case '*':
        std::cout << num1 << " " << op << " " << num2 << " " << eq << " " << (num1 * num2);
        break;
    case '/':
        std::cout << num1 << " " << op << " " << num2 << " " << eq << " " << (num1 / num2);
        break;
    case '^':
        std::cout << num1 << " " << op << " " << num2 << " " << eq << " " << pow(num1, num2);
        break;
    default:
        std::cout << "Invalid operation.";
        break;
    }
    return 0;
}